tests/commands/test_api_error.py

```diff
@@ -249,3 +249,60 @@ def test_record_tests(self):
 
         result = self.cli("record", "tests", "--session", self.session, "minitest", str(self.test_files_dir) + "/")
         self.assertEqual(result.exit_code, 0)
+
+    @responses.activate
+    @mock.patch.dict(os.environ, {"LAUNCHABLE_TOKEN": CliTestCase.launchable_token})
+    def test_all_workflow_when_server_down(self):
+        responses.add(
+            responses.GET,
+            "{base}/intake/organizations/{org}/workspaces/{ws}/verification".format(
+                base=get_base_url(),
+                org=self.organization,
+                ws=self.workspace),
+            body=ReadTimeout("error"))
+        result = self.cli("verify")
+        self.assertEqual(result.exit_code, 0)
+
+        responses.replace(
+            responses.POST,
+            "{base}/intake/organizations/{org}/workspaces/{ws}/builds".format(
+                base=get_base_url(),
+                org=self.organization,
+                ws=self.workspace),
+            body=ReadTimeout("error"))
+        result = self.cli("record", "build", "--name", "example")
+        self.assertEqual(result.exit_code, 0)
+
+        responses.replace(
+            responses.POST,
+            "{base}/intake/organizations/{org}/workspaces/{ws}/subset".format(
+                base=get_base_url(),
+                org=self.organization,
+                ws=self.workspace),
+            body=ReadTimeout("error"))
+        responses.replace(
+            responses.GET,
+            "{base}/intake/organizations/{org}/workspaces/{ws}/builds/{build}/test_sessions/{session_id}".format(
+                base=get_base_url(),
+                org=self.organization,
+                ws=self.workspace,
+                build=self.build_name,
+                session_id=self.session_id),
+            body=ReadTimeout("error"))
+        with tempfile.NamedTemporaryFile(delete=False) as rest_file:
```

Comment:
Why don't you delete temp file??

Reply:
Good catch. Thank you.

Reply:
Well, it looks like the error occurs if we try to temp file. https://app.launchableinc.com/organizations/launchableinc/workspaces/cli/data/test-sessions/2068736

Other test cases also set `delete=False` in test_api_error.py when creating temp files.

Reply:
I added the comment.